[Bug 1592263] Provide schema for SlowSQL

scholtzan opened the pull request:
See https://bugzilla.mozilla.org/show_bug.cgi?id=1592263

This PR provides a schema for `slowSQL`. The BQ schema looks like in main.4.bq:

```
...
      {
        "description": "Slow SQL queries for both the main and other threads. The execution of an SQL statement is considered slow if it takes 50ms or more on the main thread or 100ms or more on other threads. Slow SQL statements will be automatically trimmed to 1000 characters.",
        "fields": [
          {
            "fields": [
              {
                "mode": "REQUIRED",
                "name": "key",
                "type": "STRING"
              },
              {
                "mode": "REPEATED",
                "name": "value",
                "type": "INT64"
              }
            ],
            "mode": "REPEATED",
            "name": "main_thread",
            "type": "RECORD"
          },
          {
            "fields": [
              {
                "mode": "REQUIRED",
                "name": "key",
                "type": "STRING"
              },
              {
                "mode": "REPEATED",
                "name": "value",
                "type": "INT64"
              }
            ],
            "mode": "REPEATED",
            "name": "other_threads",
            "type": "RECORD"
          }
        ],
        "mode": "NULLABLE",
        "name": "slow_sql",
        "type": "RECORD"
      },
...
```

Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [x] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] If the PR comes from a fork, trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional) and review the report posted in the comments.

For glean changes:
- [ ] Update `include/glean/CHANGELOG.md`

jklukas commented:
If we can pull it off, it would be more ergonomic to model the values here as a tuple rather than as a repeated integer. So that rather than having to do `value[OFFSET(0)]`, it would be `value._0` instead. This is how we model `events` in the main ping, but I can't remember whether there's special support in MSG for event fields or if the schema transpiler itself infers this form when `items` is given as an array of types.

scholtzan commented:
I changed the schema to represent the values as tuples inferred by the schema transpiler. The bq schema looks like: 

```
{
...
        "description": "Slow SQL queries for both the main and other threads. The execution of an SQL statement is considered slow if it takes 50ms or more on the main thread or 100ms or more on other threads. Slow SQL statements will be automatically trimmed to 1000 characters.",
        "fields": [
          {
            "fields": [
              {
                "mode": "REQUIRED",
                "name": "key",
                "type": "STRING"
              },
              {
                "fields": [
                  {
                    "mode": "REQUIRED",
                    "name": "f0_",
                    "type": "INT64"
                  },
                  {
                    "mode": "REQUIRED",
                    "name": "f1_",
                    "type": "INT64"
                  }
                ],
                "mode": "REQUIRED",
                "name": "value",
                "type": "RECORD"
              }
            ],
            "mode": "REPEATED",
            "name": "main_thread",
            "type": "RECORD"
          },
          {
            "fields": [
              {
                "mode": "REQUIRED",
                "name": "key",
                "type": "STRING"
              },
              {
                "fields": [
                  {
                    "mode": "REQUIRED",
                    "name": "f0_",
                    "type": "INT64"
                  },
                  {
                    "mode": "REQUIRED",
                    "name": "f1_",
                    "type": "INT64"
                  }
                ],
                "mode": "REQUIRED",
                "name": "value",
                "type": "RECORD"
              }
            ],
            "mode": "REPEATED",
            "name": "other_threads",
            "type": "RECORD"
          }
        ],
        "mode": "NULLABLE",
        "name": "slow_sql",
        "type": "RECORD"
      },
...
```

There are already some validation tests for the `main` schema that validate `slowSQL`, I added a validation test for some more testing.
